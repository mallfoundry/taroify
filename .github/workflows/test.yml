name: CI

on:
  push:
    branches:
      - '**'

  pull_request:
    branches:
      - main

  workflow_dispatch:

jobs:
  lint:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v4

      - name: Setup Node
        uses: actions/setup-node@v4
        with:
          node-version: '20.11.1'

      - name: Install
        run: |
          yarn install:node_modules

      - name: Run linter
        run: npm run lint

  test:
    runs-on: macos-14
    steps:
      - uses: actions/checkout@v4

      - name: Setup Node
        uses: actions/setup-node@v4
        with:
          node-version: '20.11.1'

      - name: Install
        run: |
          yarn install:node_modules

      - name: Run test cases
        run: npm run test:coverage

      - name: Upload coverage to Codecov
        uses: codecov/codecov-action@v5
        with:
          token: ${{ secrets.CODECOV_TOKEN }}
          directory: ./packages/vant/test/coverage
