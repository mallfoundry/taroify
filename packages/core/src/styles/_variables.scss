// Color Palette
$black: #000;
$black-rgb: 0, 0, 0;
$white: #fff;

$gray-1: #f7f8fa;
$gray-2: #f2f3f5;
$gray-3: #ebedf0;
$gray-4: #dcdee0;
$gray-5: #c8c9cc;
$gray-6: #969799;
$gray-7: #646566;
$gray-8: #323233;

$red: #ee0a24;
$blue: #1989fa;
$cyan: #2db7f5;
$green: #07c160;
$orange: #ff976a;

$orange-dark: #ed6a0c;
$orange-light: #fffbe8;

// Base Unit
$hd: 2 !default;

// Gradient Colors
$gradient-red: linear-gradient(to right, #ff6034, #ee0a24);
$gradient-orange: linear-gradient(to right, #ffd01e, #ff8917);

// Component Colors
$primary-color: var(--blue, $blue);
$info-color: var(--cyan, $cyan);
$success-color: var(--green, $green);
$warning-color: var(--orange, $orange);
$danger-color: var(--red, $red);
$text-color: var(--gray-8, $gray-8);
$text-color-2: var(--gray-6, $gray-6);
$text-color-3: var(--gray-5, $gray-5);
$text-link-color: #576b95;
$active-color: var(--gray-2, $gray-2);
$active-opacity: 0.7;
$disabled-opacity: 0.5;
$background-color: var(--gray-1, $gray-1);
$background-color-light: var(--white, $white);
$text-link-color: #576b95;

// Padding
$padding-base: 4px * $hd;
$padding-xs: $padding-base * 2;
$padding-sm: $padding-base * 3;
$padding-md: $padding-base * 4;
$padding-lg: $padding-base * 6;
$padding-xl: $padding-base * 8;

// Height
$height-xs: 16px * $hd;
$height-sm: 18px * $hd;
$height-md: 20px * $hd;
$height-lg: 22px * $hd;

$font-weight-bold: 500;

// Font
$font-size-xs: 10px * $hd;
$font-size-sm: 12px * $hd;
$font-size-md: 14px * $hd;
$font-size-lg: 16px * $hd;
// Line Height
$line-height-xs: 14px * $hd;
$line-height-sm: 18px * $hd;
$line-height-md: 20px * $hd;
$line-height-lg: 22px * $hd;

$base-font-family: -apple-system, blinkmacsystemfont, 'Helvetica Neue',
helvetica, segoe ui, arial, roboto, 'PingFang SC', 'miui', 'Hiragino Sans GB',
'Microsoft Yahei', sans-serif;

$price-integer-font-family: avenir-heavy, pingfang sc, helvetica neue, arial, sans-serif;

// Animation
$animation-duration-base: 0.3s;
$animation-duration-fast: 0.2s;
$animation-timing-function-enter: ease-out;
$animation-timing-function-exit: ease-in;

// Border
$border-color: var(--gray-3, $gray-3);
$border-width-base: 1px * $hd;
$border-radius-sm: 2px * $hd;
$border-radius-md: 4px * $hd;
$border-radius-lg: 8px * $hd;
$border-radius-max: 999px;
