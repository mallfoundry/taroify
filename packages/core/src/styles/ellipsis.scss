@import './prefix';
@import './mixins/ellipsis';

.#{$component-prefix}ellipsis {
  @include ellipsis();
}

.#{$component-prefix}ellipsis--l2 {
  @include ellipsis(2);
}

.#{$component-prefix}ellipsis--l3 {
  @include ellipsis(3);
}
