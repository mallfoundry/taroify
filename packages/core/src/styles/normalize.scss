@import "./prefix";
@import "./variables";

[class*="#{$component-prefix}"] {
  -webkit-tap-highlight-color: transparent;
}

a {
  text-decoration: none;
}

input,
button,
textarea {
  font: inherit;
  color: inherit;
}

a,
input,
button,
textarea,
[class*="#{$component-prefix}"] {
  &:focus {
    outline: none;
  }
}

ol,
ul {
  padding: 0;
  margin: 0;
  list-style: none;
}
