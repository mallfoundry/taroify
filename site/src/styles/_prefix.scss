$taroify-prefix: "t" + "a" + "r" + "o" + "i" + "f" + "y" + "-";
$component-prefix: $taroify-prefix;
