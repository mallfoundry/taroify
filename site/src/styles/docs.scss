@import "./prefix";
@import "./variables";

.#{$component-prefix}docs {
  &-container {
    box-sizing: border-box;
    padding-left: 220px;
    overflow: hidden;

    &-with-simulator {
      padding-right: 384px;
    }
  }

  @media (min-width: 1680px) {
    &-container {
      width: 1680px;
      margin: 0 auto;
    }
  }

  &-content {
    padding: 30px;
    overflow: hidden;

    a {
      margin: 0 1px;
      color: $vant-blue;
      text-decoration: none;

      &:hover {
        color: darken($vant-blue, 10%);
      }

      &:active {
        color: darken($vant-blue, 20%);
      }
    }

    h1,
    h2,
    h3,
    h4,
    h5,
    h6 {
      font-weight: normal;
      line-height: 1.5;
      color: $vant-black;

      &[id] {
        cursor: pointer;
      }
    }

    h1 {
      margin: 0 0 30px;
      font-size: 30px;
      cursor: default;
    }

    h2 {
      margin: 45px 0 20px;
      font-size: 25px;
    }

    h3 {
      margin-bottom: 16px;
      font-size: 18px;
      font-weight: 600;
    }

    h4 {
      margin: 24px 0 12px;
      font-size: 16px;
      font-weight: 600;
    }

    h5 {
      margin: 24px 0 12px;
      font-size: 15px;
      font-weight: 600;
    }

    p {
      margin: 16px 0;
      font-size: 15px;
      line-height: 26px;
      color: $vant-text-color;
    }

    table {
      width: 100%;
      margin-top: 12px;
      font-size: 14px;
      line-height: 1.5;
      color: $vant-text-color;
      border-collapse: collapse;

      th {
        padding: 8px 10px;
        font-weight: 600;
        text-align: left;

        &:first-child {
          padding-left: 0;
        }

        &:last-child {
          padding-right: 0;
        }
      }

      td {
        padding: 8px;
        border-top: 1px solid $vant-code-background-color;

        &:first-child {
          padding-left: 0;
          // version tag
          code {
            padding: 2px 6px;
            margin: 0;
            font-size: 11px;
            font-weight: 600;
            color: $vant-blue;
            background-color: fade($vant-blue, 10%);
            border-radius: 20px;
          }
        }

        &:last-child {
          padding-right: 0;
        }
      }

      em {
        font-family: $vant-code-font-family;
        font-size: 14px;
        font-style: normal;
        color: $vant-green;
      }
    }

    ul, ol {
      padding: 0;
      list-style: none;
    }

    ul li,
    ol li {
      position: relative;
      padding-left: 15px;
      margin: 5px 0 5px 10px;
      font-size: 15px;
      line-height: 26px;
      color: $vant-text-color;

      &::before {
        position: absolute;
        top: 0;
        left: 0;
        box-sizing: border-box;
        width: 6px;
        height: 6px;
        margin-top: 10px;
        content: '';
        border: 1px solid $vant-dark-grey;
        border-radius: 50%;
      }
    }

    hr {
      margin: 30px 0;
      border: 0 none;
      border-top: 1px solid #eee;
    }

    p > code,
    li > code,
    table code {
      display: inline;
      padding: 2px 5px;
      margin: 0 2px;
      font-family: inherit;
      font-size: 14px;
      word-break: keep-all;
      background-color: $vant-background-color;
      border-radius: 4px;
      -webkit-font-smoothing: antialiased;
    }

    p > code {
      font-size: 14px;
    }

    section {
      padding: 24px;
      overflow: hidden;
    }

    blockquote {
      padding: 16px;
      margin: 16px 0 0;
      background-color: #ecf9ff;
      border-radius: $vant-border-radius;

      p {
        margin: 0;
      }
    }

    img {
      width: 100%;
      margin: 16px 0;
      border-radius: $vant-border-radius;
    }
  }

  &-card {
    padding: 24px;
    margin-bottom: 24px;
    background-color: #fff;
    border-radius: $vant-border-radius;
    box-shadow: 0 8px 12px #ebedf0;

    h3 {
      margin-top: 0;
    }
  }
}
