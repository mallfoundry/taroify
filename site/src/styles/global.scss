@import "./variables";

body {
  min-width: 1100px;
  font-family: 'Open Sans', -apple-system, BlinkMacSystemFont, 'Helvetica Neue',
  Helvetica, Segoe UI, Arial, Roboto, 'PingFang SC', 'miui',
  'Hiragino Sans GB', 'Microsoft Yahei', sans-serif;
  background-color: $vant-background-color;
}
