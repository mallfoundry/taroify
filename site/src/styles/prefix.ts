const COMPONENT_PREFIX = "\u0074\u0061\u0072\u006f\u0069\u0066\u0079-"

export function prefixClassname(component: string) {
  return `${COMPONENT_PREFIX}${component}`
}
