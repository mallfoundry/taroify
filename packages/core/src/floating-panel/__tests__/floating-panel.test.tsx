import * as React from "react"
import { getSystemInfoSync } from "@tarojs/taro"
import { render } from "@testing-library/react"
import FloatingPanel from "../index"
import { prefixClassname } from "../../styles"
import { touchEvent, sleep } from "../../../test"

describe("<FloatingPanel />", () => {
  it("should minHeight 100 and maxHeight 0.6 innerHeight when anchors props do not", async () => {
    const { asFragment, container } = render(<FloatingPanel>内容</FloatingPanel>)

    expect(asFragment()).toMatchSnapshot()

    const el = container.querySelector(`.${prefixClassname("floating-panel")}`) as HTMLDivElement

    const height = Math.round(getSystemInfoSync().windowHeight * 0.6)

    expect(el.style.height).toBe(`${height}px`)

    expect(el.style.transform).toContain(`${height}px`)
  })

  it("should drag adsorption effect when anchors props is [100, 200, 400]", async () => {
    const { container, asFragment } = render(
      <FloatingPanel anchors={[100, 200, 400]}>内容</FloatingPanel>,
    )

    const el = container.querySelector(`.${prefixClassname("floating-panel")}`) as HTMLDivElement

    expect(asFragment()).toMatchSnapshot()

    expect(el.style.height).toBe("400px")

    // drag 10
    touchEvent(el, 0, 10)
    await sleep(0)
    expect(el.style.transform).toContain("-100px")

    // drag -49
    touchEvent(el, 0, -49)
    await sleep(0)
    expect(el.style.transform).toContain("-100px")

    // drag -199
    touchEvent(el, 0, -199)
    await sleep(0)
    expect(el.style.transform).toContain("-200px")

    // drag -200
    touchEvent(el, 0, -200)
    await sleep(0)
    expect(el.style.transform).toContain("-400px")

    // drag -500
    touchEvent(el, 0, -500)
    await sleep(0)
    expect(el.style.transform).toContain("-400px")
  })

  it("should emit height-change when height change in anchors", async () => {
    const onChange = jest.fn()
    const { container } = render(
      <FloatingPanel anchors={[100, 200, 400]} handleChange={onChange}>
        内容
      </FloatingPanel>,
    )
    const el = container.querySelector(`.${prefixClassname("floating-panel")}`) as HTMLDivElement
    touchEvent(el, 0, -199)
    await sleep(0)
    expect(el.style.transform).toContain("-200px")
    expect(onChange).toBeCalledWith(299)
  })

  it("should only drag header when allowDraggingContent is false", async () => {
    const onChange = jest.fn()
    const { container } = render(
      <FloatingPanel anchors={[100, 200, 400]} contentDraggable={false} handleChange={onChange}>
        内容
      </FloatingPanel>,
    )
    const el = container.querySelector(
      `.${prefixClassname("floating-panel__content")}`,
    ) as HTMLDivElement
    touchEvent(el, 0, -199)
    await sleep(0)
    expect(onChange).not.toHaveBeenCalled()

    const elHeader = container.querySelector(
      `.${prefixClassname("floating-panel__header")}`,
    ) as HTMLDivElement
    touchEvent(elHeader, 0, -199)
    await sleep(0)
    expect(onChange).toBeTruthy()
  })
})
