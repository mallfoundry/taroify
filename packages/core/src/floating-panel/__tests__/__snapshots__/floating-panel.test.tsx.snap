// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FloatingPanel /> should drag adsorption effect when anchors props is [100, 200, 400] 1`] = `
<DocumentFragment>
  <taro-view-core
    class="taroify-floating-panel"
    style="height: 400px; transform: translateY(calc(100% + -100px)); transition: transform 0.3s cubic-bezier(0.18, 0.89, 0.32, 1.28);"
  >
    <taro-view-core
      class="taroify-floating-panel__header"
      data-id="floating-panel__header"
    >
      <taro-view-core
        class="taroify-floating-panel__header-bar"
      />
    </taro-view-core>
    <taro-scroll-view-core
      class="taroify-floating-panel__content"
      scroll-y="true"
    >
      内容
      <taro-view-core
        class="taroify-safe-area taroify-safe-area--bottom"
      />
    </taro-scroll-view-core>
  </taro-view-core>
</DocumentFragment>
`;

exports[`<FloatingPanel /> should minHeight 100 and maxHeight 0.6 innerHeight when anchors props do not 1`] = `
<DocumentFragment>
  <taro-view-core
    class="taroify-floating-panel"
    style="height: 0px; transform: translateY(calc(100% + 0px)); transition: transform 0.3s cubic-bezier(0.18, 0.89, 0.32, 1.28);"
  >
    <taro-view-core
      class="taroify-floating-panel__header"
      data-id="floating-panel__header"
    >
      <taro-view-core
        class="taroify-floating-panel__header-bar"
      />
    </taro-view-core>
    <taro-scroll-view-core
      class="taroify-floating-panel__content"
      scroll-y="true"
    >
      内容
      <taro-view-core
        class="taroify-safe-area taroify-safe-area--bottom"
      />
    </taro-scroll-view-core>
  </taro-view-core>
</DocumentFragment>
`;
